src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -176,10 +174,21 @@
              "The provided authorization service config is not a list.")))
   (doseq [rule config]
     (validate-auth-config-rule! rule))
+  (doseq [[name rules] (group-by :name config)]
+    (when-not (= 1 (count rules))
+      (throw (IllegalArgumentException.
+              (str "Duplicate rules named '" name "'. "
+                   "Rules must be uniquely named.")))))
   config)
 
 (schema/defn transform-config :- rules/Rules
-  "Transforms the authorization service config into a list of Rules that work
-  with the authorization code."
+  "Transforms the (validated) authorization service config into a list of Rules
+   that work with the authorization code. Assumes config has been validated via
+   `validate-auth-config!`. A warning is logged if the rules in the config are
+   not in ascending sort order."
   [config]
-  (map config->rule config))
+  (let [parsed (map config->rule config)
+        trim-fn #(select-keys % [:sort-order :name])]
+    (when-not (= (map trim-fn config) (map trim-fn (rules/sort-rules parsed)))
+      (log/warn "Rules in configuration file not in ascending sort order."))
```

Comment:
To help a user troubleshoot this further, would be nice to have the warning message indicate the first instance of a rule that is out of order - by rule name - depending upon how hard that would be to do.